src/logging.rs

```diff
@@ -0,0 +1,51 @@
+//! Provides debug logging.
+//!
+//! WARNING: Logging leaks secret information and is provided for debugging purposes only.
+//!
+//! DO NOT ENABLE LOGGING IN PRODUCTION SYSTEMS.
+//!
+
+use std::fs::File;
+use std::sync::Mutex;
+
+#[allow(dead_code)] // When danger_leak_secret_material not enabled.
+pub static LOGFILE: Mutex<Option<File>> = Mutex::new(None);
+
+#[allow(dead_code)] // When danger_leak_secret_material not enabled.
+pub const CRYPTO_DEBUG_FILE: &str = "/tmp/rust-secp256k1.log";
+
+/// Logs a format string to `CRYPTO_DEBUG_FILE`.
+#[cfg(feature = "danger_leak_secret_material")]
+#[macro_export]
+macro_rules! log {
```

Comment:
In 8cb37e093bf91bd1f621af5f35ca473153bd5884:

This should have a different name to make it clearer that it's a debug-only thing. Maybe `trace_log` or just `trace`.

Reply:
Could still be confused with `trace` macro: https://docs.rs/log/latest/log/macro.trace.html. What about `leak!` if we use the `pub(crate)` way of exporting the macro usage could be `logging::leak!("...")`?

Reply:
I'll play with both and see how they feel. Cheers.